packages/@aws-cdk/core/lib/cfn-resource.ts

```diff
@@ -15,6 +15,10 @@ import { TagManager } from './tag-manager';
 import { Tokenization } from './token';
 import { capitalizePropertyNames, ignoreEmpty, PostResolveToken } from './util';
 import { FeatureFlags } from './feature-flags';
+import { Stack } from './stack';
+
+type Element = CfnResource | Stack;
```

Comment:
I'm not sure this will play nice with jsii.

Reply:
I copied that from `deps.ts` since it seemed convenient, so it looks like it may work OK. If you confirm you want it gone, I'm happy to remove.

Reply:
I'm still working on removing the `Element` types - I'm struggling a bit with not letting the type bleed out of `deps.ts` into the other interfaces.

Reply:
I found this a useful reference to explain the jsii issue, btw:
https://aws.github.io/jsii/specification/2-type-system/#type-unions

> Type Unions
>
>In certain cases, several different kinds of values are acceptable for a given parameter or return type. TypeScript models those cases using type unions, which are represented as TypeA | TypeB. The jsii type model supports those, however most other statically typed languages do not have such a concept, making those parameters or return values difficult to use from those languages, as the value has to be declared using the most generic reference type available (for example, in Java, those are returned as java.lang.Object).
>
>When used as inputs (parameters, or properties of a struct), it may be possible to generate method overloads that will allow for a convenient API in languages that support overloads.
>
>In general however, type unions are discouraged and should only be used when there is no alternative way to model the API.

Reply:
Okay, I had to be a bit more aggressive about replacing `Element` than I initially hoped - I had to remove it entirely, including from `deps.ts` and replace it with a token interface to represent the same, but without the union type. I'm not entirely sure how this interface will wind up represented in docs and target languages so I may need help putting some polish this part - if you like the approach anyway.